invenio_files_rest/models.py

```diff
@@ -1306,6 +1306,25 @@ def delete(cls, bucket, key):
             return cls.create(bucket, key)
         return None
 
+    @classmethod
+    def remove_many(cls, version_ids, bucket):
```

Comment:
minor: 
```suggestion
    def delete_many(cls, version_ids, bucket):
```

Reply:
There are 2, remove and delete. Remove is hard delete (the one causing the slowness) while delete is soft delete. So far we only need the multiple delete for hard deletion that's why I name it remove_many.